libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -574,14 +574,25 @@ - (void)beginUserAgentFlow {
                         swapOK = YES;
                         self.credentials.jwt = nil;
                     }
+                    else {
+                        if ([[dict valueForKey:@"error"] isEqual:kSFOAuthErrorTypeInvalidGrant]) {
+                            NSError *error = [NSError errorWithDomain:kSFOAuthEndPointToken
```

Comment:
To follow the pattern of OAuth errors returned to the client, we should:
1. Generate errors from `[SFOAuthCoordinator errorWithType:description:]`
2. Create the equivalent `NSString` constant type to go with the `kSFOAuthError` key in the user dictionary, and add the conditional logic to generate that error type/code—the errorWithType implementation mentioned above will make the pattern clear.